upgrade/v122to123/upgrade.go

```diff
@@ -42,6 +42,24 @@ func upgradeBackups(namespace string, lhClient *lhclientset.Clientset) (err erro
 		return err
 	}
 
+	engines, err := lhClient.LonghornV1beta2().Engines(namespace).List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+	volumeNameToEngines := make(map[string][]longhorn.Engine)
+	for _, e := range engines.Items {
+		volumeNameToEngines[e.Labels[types.LonghornLabelVolume]] = append(volumeNameToEngines[e.Labels[types.LonghornLabelVolume]], e)
+	}
+
+	volumes, err := lhClient.LonghornV1beta2().Volumes(namespace).List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+	volumeMap := make(map[string]longhorn.Volume)
+	for _, v := range volumes.Items {
+		volumeMap[v.Name] = v
+	}
```

Comment:
nit: Do you think we could have this part delayed init until there is a 2-engines case happen in the default clause, because the below backup loop is not parallel.

Reply:
Could you give more details about how we can delay?

On the other hand, volume CR listing should be fast and doesn't affect much to overall time.

Reply:
Have a function closure to get volume-map (init once) outside the backup loop, then call it the default clause?